packages/dma-library/src/strategies/morphoblue/multiply/adjust.ts

```diff
@@ -0,0 +1,353 @@
+  )
+  const mappedArgs = {
+    ...args,
+    collateralTokenSymbol,
+    quoteTokenSymbol: debtTokenSymbol,
+    collateralAmount: args.collateralAmount.shiftedBy(args.collateralTokenPrecision),
+  }
+
+  const mappedPosition: MinimalPosition = {
+    collateralAmount: args.position.collateralAmount.shiftedBy(args.collateralTokenPrecision),
+    debtAmount: args.position.debtAmount.shiftedBy(args.quoteTokenPrecision),
+    riskRatio: args.position.riskRatio,
+    marketParams: {
+      loanToken: args.position.marketParams.loanToken,
+      collateralToken: args.position.marketParams.collateralToken,
+    },
+  }
+
+  // Simulate adjust
+  const riskIsIncreasing = false
+  const simulatedAdjustment = await simulateAdjustment(
+    mappedArgs,
+    dependencies,
+    mappedPosition,
+    riskIsIncreasing,
+    oraclePrice,
+    collateralTokenSymbol,
+    debtTokenSymbol,
+  )
+
+  // Get swap data
+  const { swapData, collectFeeFrom, preSwapFee } = await getSwapData(
+    mappedArgs,
+    args.position,
+    dependencies,
+    simulatedAdjustment,
+    riskIsIncreasing,
+    positionType,
+    collateralTokenSymbol,
+    debtTokenSymbol,
+  )
+
+  // Build operation
+  const operation = await buildOperation(
+    args,
+    dependencies,
+    simulatedAdjustment,
+    swapData,
+    riskIsIncreasing,
+  )
+
+  // Prepare payload
+  return prepareMorphoMultiplyDMAPayload(
+    args,
+    dependencies,
+    simulatedAdjustment,
+    operation,
+    swapData,
+    collectFeeFrom,
+    preSwapFee,
+    riskIsIncreasing,
+    args.position,
+    collateralTokenSymbol,
+    debtTokenSymbol,
+  )
+}
+
+async function buildOperation(
+  args: MorphoAdjustMultiplyPayload,
+  dependencies: MorphoMultiplyDependencies,
+  simulatedAdjust: Domain.ISimulationV2 & Domain.WithSwap,
+  swapData: SwapData,
+  riskIsIncreasing: boolean,
+): Promise<IOperation> {
+  /** Not relevant for Ajna */
+  const debtTokensDeposited = ZERO
+  const borrowAmount = simulatedAdjust.delta.debt.minus(debtTokensDeposited)
+  const collateralTokenSymbol = simulatedAdjust.position.collateral.symbol.toUpperCase()
+  const debtTokenSymbol = simulatedAdjust.position.debt.symbol.toUpperCase()
+  const fee = SwapUtils.feeResolver(collateralTokenSymbol, debtTokenSymbol, {
+    isIncreasingRisk: riskIsIncreasing,
+    isEarnPosition: SwapUtils.isCorrelatedPosition(collateralTokenSymbol, debtTokenSymbol),
+  })
+  const swapAmountBeforeFees = simulatedAdjust.swap.fromTokenAmount
+  const collectFeeFrom = SwapUtils.acceptedFeeTokenBySymbol({
+    fromTokenSymbol: riskIsIncreasing
+      ? simulatedAdjust.position.debt.symbol
+      : simulatedAdjust.position.collateral.symbol,
+    toTokenSymbol: riskIsIncreasing
+      ? simulatedAdjust.position.collateral.symbol
+      : simulatedAdjust.position.debt.symbol,
+  })
+
+  const network = await getNetwork(dependencies.provider)
+
+  if (riskIsIncreasing) {
+    const riskUpMultiplyArgs: MorphoBlueAdjustRiskUpArgs = {
```

Comment:
I'd extract these 2 pieces of code into separate functions, the `buildOperation` function is quite big